configure.go

```diff
@@ -0,0 +1,135 @@
+package configure
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"reflect"
+	"strings"
+)
+
+const (
+	structTagKey   = "config"
+	requiredTagKey = "required"
+)
+
+var (
+	ErrStringsOnly            = errors.New("Only string values are allowed in a config struct.")
+	ErrNotReference           = errors.New("The config struct must be a pointer.")
+	ErrStructOnly             = errors.New("Config object must be a struct.")
+	ErrNoTagValue             = errors.New("Config object attributes must have a 'config' tag value.")
+	ErrFlagParsed             = errors.New("Flags should not be used in conjuction, do not flag.Parse() before Configure")
```

Comment:
Minor: I was confused by the first half of the error ("Flags should not be used in conjunction") - sounds like I shouldn't use flags/json together (which might also be true, but is not what this error means).